pkg/autodiscovery/integration/config.go

```diff
@@ -38,16 +38,17 @@ const (
 
 // Config is a generic container for configuration files
 type Config struct {
-	Name          string       `json:"check_name"`     // the name of the check
-	Instances     []Data       `json:"instances"`      // array of Yaml configurations
-	InitConfig    Data         `json:"init_config"`    // the init_config in Yaml (python check only)
-	MetricConfig  Data         `json:"metric_config"`  // the metric config in Yaml (jmx check only)
-	LogsConfig    Data         `json:"logs"`           // the logs config in Yaml (logs-agent only)
-	ADIdentifiers []string     `json:"ad_identifiers"` // the list of AutoDiscovery identifiers (optional)
-	Provider      string       `json:"provider"`       // the provider that issued the config
-	Entity        string       `json:"-"`              // the id of the entity (optional)
-	ClusterCheck  bool         `json:"cluster_check"`  // cluster-check configuration flag
-	CreationTime  CreationTime `json:"-"`              // creation time of service
+	Name            string       `json:"check_name"`     // the name of the check
+	Instances       []Data       `json:"instances"`      // array of Yaml configurations
+	InitConfig      Data         `json:"init_config"`    // the init_config in Yaml (python check only)
+	MetricConfig    Data         `json:"metric_config"`  // the metric config in Yaml (jmx check only)
+	LogsConfig      Data         `json:"logs"`           // the logs config in Yaml (logs-agent only)
+	ADIdentifiers   []string     `json:"ad_identifiers"` // the list of AutoDiscovery identifiers (optional)
+	Provider        string       `json:"provider"`       // the provider that issued the config
+	Entity          string       `json:"-"`              // the id of the entity (optional)
+	ClusterCheck    bool         `json:"cluster_check"`  // cluster-check configuration flag
+	EndpointsChecks []Config     `json:"-"`              // endpoints check configs related to a service
```

Comment:
Would it be possible to avoid doing this? (storing a slice of `[]Config` inside  a `Config`)

The `Integration.Config` object is supposed to be quite generic (used by the logs agent for example) so we try to keep business logic minimal here. The config provider can return the list of `Config` Directly instead of storing it into `EndpointsChecks` ?

Reply:
The reason for storing endpoints check configs in the kube service config is because we don't have auto-discovery listeners for endpoints, and since endpoints checks are controlled (enabled/disabled/modified) by users via the kube service annotations we're relying on the kube service listeners to report any changes. Endpoints checks wouldn't be resolved if we return them directly in the kube service config provider and this makes sense because endpoints checks config provider is used by node agents not the cluster agent.
We discussed with @hkaj the need to revamp (in a next PR) the whole endpoints checks feature by completely separating it from service checks (in the dca dispatcher logic also) and we will be able to get rid of `EndpointsChecks` field then.
So WDYT? I would be happy to discuss it further :)

Reply:
OK got it 👍 
Indeed we'll need to keep a map to keep the mapping between `kube_service` and `endpoints` in the same fashion we do for `services` and `configs` in the AD, let's go for the simple solution for now.